From 4f571944b75ad512078ca5ca2703a159d405ad95 Mon Sep 17 00:00:00 2001
From: Adam Jackson <ajax@redhat.com>
Date: Fri, 23 Jul 2021 16:13:45 -0400
Subject: [PATCH] glx: disable the indirect fallback in CreateContextAttribs

If your app cares enough to use CreateContextAttribs it's probably not
going to be happy with the pre-GL-1.5 indirect experience.
---
 src/glx/create_context.c | 13 ++++++-------
 1 file changed, 6 insertions(+), 7 deletions(-)

diff --git a/src/glx/create_context.c b/src/glx/create_context.c
index c44d579c954..f31d466a732 100644
--- a/src/glx/create_context.c
+++ b/src/glx/create_context.c
@@ -109,16 +109,15 @@ glXCreateContextAttribsARB(Display *dpy, GLXFBConfig config,
       gc = psc->vtable->create_context_attribs(psc, cfg, share, num_attribs,
 					       (const uint32_t *) attrib_list,
 					       &dummy_err);
-   }
-
-   if (gc == NULL) {
+   } 
 #ifdef GLX_USE_APPLEGL
+   else if (gc == NULL) {
       gc = applegl_create_context(psc, cfg, share, 0);
-#else
-      gc = indirect_create_context_attribs(psc, cfg, share, num_attribs,
-              (const uint32_t *) attrib_list,
-              &dummy_err);
 #endif
+   else if (!direct) {
+      gc = indirect_create_context_attribs(psc, cfg, share, num_attribs,
+                                           (const uint32_t *) attrib_list,
+                                           &dummy_err);
    }
 
    xid = xcb_generate_id(c);
-- 
GitLab

